### DISCUSSION TITLE
Installing SpacePy on fresh Anaconda on Windows

### DISCUSSION THREAD

**Discussion by Yoshi2112** [spacepy/spacepy#465] [Support and Q&A]
Hi all, not sure if this is the appropriate place to put this, but it's worth a shot. Due to an issue with conda (accidentally closed the cmd window during a python update) I had to reinstall everything from scratch. However, following the steps on in the SpacePy docs, every attempt I've made to install SpacePy has failed (both through pip install and by setup.py install on the downloaded repo). 

In both instances, it appears as though it's a Fortran issue, yet both gfortran and gcc are in my Windows 10 path and accessible through the cmd. I have attached the piped output from both attempts (though it doesn't include everything - I can copy/paste more if required:

https://gofile.io/d/d6EcTI

It seems like a strange error - my SpacePy install was working perfectly before I had to reinstall everything. But I'm also aware that many things have probably been updated since I last installed it. Unfortunately I'm not a programmer - I know just enough to get myself into trouble. Any suggestions would be much appreciated!

UPDATE: I just realised there was an .exe that I could use, and it successfully installed! SpacePy is now confirmed working on my system. I am still confused as to why I couldn't install it via pip or source. Not sure if perhaps it was just a bug on my own personal machine.

**Comment by jtniehof** [comment]
Thanks for filing the issue. I've converted it to a discussion and edited the subject so hopefully it'll help people find it...if we wind up with bug fixes or documentation additions we can file issue to hold those.

Bottom line is I'm glad you got things working. I'm also in the process of preparing another release and pretty much every release includes installer improvements. `pip` should normally download and install the binary wheel on Windows, so the fact that it tried to build indicates something made it think it wasn't compatible. It's possible that your Anaconda install is running a newer version of Python; we only had binary wheels up through 3.7 since 3.8 wasn't out when 0.2.1 was released.

As far as the build goes, I can't get much detail from the output you posted; the pip install has no information other than "error" and the build from source stops right where the build proper starts. Some details are visible in the [script we use to set up a Windows build system](https://github.com/spacepy/spacepy/blob/master/developer/scripts/win_build_system_setup.cmd) but the key is to use Anaconda to install `m2w64-gcc-fortran` and `libpython` before starting the build. This is documented in the [Windows install docs](https://spacepy.github.io/install_windows.html#compiling).

**Reply by Yoshi2112** [reply]
Hi Jon,

Thanks for the reply and the conversion. I wasn't sure how to pipe the cmd output to file (it piped some, but not all?). For the record, yes the new Anaconda uses python 3.8 (I thought that might cause some issues). I did make sure to install m2w64-gcc-fortran and libpython though, as I followed the Windows install docs.

Cheers,

Josh

**Reply by jtniehof** [reply]
A lot of the information goes to stderr, so the redirect has to be something like:
```
python setup.py install > output.txt 2>&1
```
Glad you found the docs (always a question of whether people are able to find the relevant bits in a sea of docs.) The 3.8 would be why the binary wheel wasn't grabbed. There will be a Windows binary wheel for 3.8 in 0.2.2 (and 3.9, hopefully.)

**Reply by jtniehof** [reply]
Closing this as there have been a lot of installer updates, so this is probably no longer useful for folks.
